amtl/am-string.h

```diff
@@ -115,11 +115,28 @@ class AString
            memcmp(other.chars(), chars(), length()) == 0;
   }
 
+  friend AString operator +(const AString &a, const AString &b) {
```

Comment:
I've never seen the `friend` keyword placed on a definition before, only on declarations. What does it do here?